### PR TITLE
Upgrade embedded Kotlin to 1.3.30

### PR DESCRIPTION
**Pull request by bamboo**
Signed-off-by: Rodrigo B. de Oliveira <rodrigo@gradle.com>

### CODE REVIEW

**Comment by eskatos** [comment]
`AbstractSmokeTest` should be updated too

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy [hunk 1]
```diff
@@ -80,7 +80,7 @@ abstract class AbstractSmokeTest extends Specification {
         static androidGradle = Versions.of("3.2.1", androidGradle3x)
 
         // https://search.maven.org/search?q=g:org.jetbrains.kotlin%20AND%20a:kotlin-project&core=gav
-        static kotlin = Versions.of('1.2.21', '1.2.31', '1.2.41', '1.2.51', '1.2.61', '1.2.71', '1.3.0', '1.3.11', '1.3.21')
+        static kotlin = Versions.of('0.2.21', '1.2.31', '1.2.41', '1.2.51', '1.2.61', '1.2.71', '1.3.0', '1.3.11', '1.3.21', '1.3.30')
```
Comment (eskatos) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy]: :x: typo
```suggestion
        static kotlin = Versions.of('1.2.21', '1.2.31', '1.2.41', '1.2.51', '1.2.61', '1.2.71', '1.3.0', '1.3.11', '1.3.21', '1.3.30')
```

File: subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/KotlinCompiler.kt [hunk 2]
```diff
@@ -295,6 +298,7 @@ private
 fun compilerConfigurationFor(messageCollector: MessageCollector): CompilerConfiguration =
     CompilerConfiguration().apply {
         put(CLIConfigurationKeys.MESSAGE_COLLECTOR_KEY, messageCollector)
+        put(JVMConfigurationKeys.USE_FAST_CLASS_FILES_READING, true)
```
Comment (eskatos) [inline subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/KotlinCompiler.kt]: ❓ I think it is the default already, is it?

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy (hunk 1 above)
Comment (bamboo) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy]: Thanks!

File: subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/KotlinCompiler.kt (hunk 2 above)
Comment (bamboo) [reply subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/KotlinCompiler.kt]: It doesn't seem to be the default value for the code path we use. I can see code setting it to `true` based on the command line arguments but not when control flows from the API directly.  So I'd rather keep it explicitly enabled.

Comment (eskatos) [reply subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/KotlinCompiler.kt]: Right, I mixed my mind with `-Xuse-old-class-files-reading` and the "current class files reading" being the default, see https://github.com/gradle/kotlin-dsl/pull/1004/files#diff-662c4a38f3704b9931869cb943a3c2afR40. Now it's another "fast class files reading" 🤷‍♂️ 

FTR, tests sensible to this were the ones involving parameter names metadata reading and our API jar generation.

**Comment by bamboo** [comment]
@eskatos Done.

**Pull request merged**